features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.view/public/js/device-view.js

```diff
@@ -0,0 +1,204 @@
+                        policyObjectToView["priorityId"] = policyObjectFromRestEndpoint["priorityId"];
+                        policyObjectToView["name"] = policyObjectFromRestEndpoint["policyName"];
+                        policyObjectToView["platform"] = policyObjectFromRestEndpoint["profile"]["deviceType"]["name"];
+                        policyObjectToView["ownershipType"] = policyObjectFromRestEndpoint["ownershipType"];
+                        policyObjectToView["compliance"] = policyObjectFromRestEndpoint["compliance"];
+
+                        if (policyObjectFromRestEndpoint["active"] == true && policyObjectFromRestEndpoint["updated"] == true) {
+                            policyObjectToView["status"] = "Active/Updated";
+                        } else if (policyObjectFromRestEndpoint["active"] == true && policyObjectFromRestEndpoint["updated"] == false) {
+                            policyObjectToView["status"] = "Active";
+                        } else if (policyObjectFromRestEndpoint["active"] == false && policyObjectFromRestEndpoint["updated"] == true) {
+                            policyObjectToView["status"] = "Inactive/Updated";
+                        } else if (policyObjectFromRestEndpoint["active"] == false && policyObjectFromRestEndpoint["updated"] == false) {
+                            policyObjectToView["status"] = "Inactive";
+                        }
+                        // push view-objects to list
+                        policyListToView.push(policyObjectToView);
+                    }
+                    viewModel.policies = policyListToView;
+                    var content = template(viewModel);
+                    $("#policy-list-container").html(content);
+                    $('#policy-table').datatables_extended();
+                }
+            };
+            invokerUtil.get(serviceURL,
+                            successCallback, function (message) {
+                        console.log(message);
+                    });
+        });
+
+    }
+
+    function loadPolicyCompliance() {
+        var policyCompliance = $("#policy-view");
+        var policySrc = policyCompliance.attr("src");
+        var deviceId = policyCompliance.data("device-id");
+        var deviceType = policyCompliance.data("device-type");
+        var activePolicy = null;
+
+        $.template("policy-view", policySrc, function (template) {
+            var serviceURLPolicy = "/devicemgt_admin/policies/" + deviceType + "/" + deviceId + "/active-policy"
+            var serviceURLCompliance = "/devicemgt_admin/policies/" + deviceType + "/" + deviceId;
+
+            var successCallbackCompliance = function (data) {
+                var viewModel = {};
+                viewModel.policy = activePolicy;
+                viewModel.deviceType = deviceType;
+                if (data != null && data.complianceFeatures != null && data.complianceFeatures != undefined && data.complianceFeatures.length > 0) {
```

Comment:
I think you can simplify here using ,
if (data && data.complianceFeatures  && data.complianceFeatures.length > 0)